src/application.coffee

```diff
@@ -463,10 +467,12 @@ getRemoteState = (uuid, apiKey) ->
 		console.error("Failed to get state for device #{uuid}. #{err}")
 		throw err
 
+# TODO: Actually store and use app.environment and app.config separately
 parseEnvAndFormatRemoteApps = (remoteApps, uuid, apiKey) ->
 	appsWithEnv = _.mapValues remoteApps, (app, appId) ->
 		utils.extendEnvVars(app.environment, uuid, appId)
-		.then (fullEnv) ->
+		.then (extendedEnv) ->
+			fullEnv = _.merge({}, extendedEnv, app.config)
 			env = _.omit(fullEnv, _.keys(specialActionEnvVars))
```

Comment:
You can drop these omits now and just use `extendedEnv`, since these omits are just trying to get rid of the things that are only in app.config - so if you use the version before you add the app.config stuff you'll be fine